### PR TITLE
v23.0.0 release

### PR DESCRIPTION
**Pull request by ErichDonGubler**
**Connections**

- Follows the process currently in `RELEASE.md`.

**Description**

\- 

**Testing**

\-

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
@gfx-rs/naga and @gfx-rs/wgpu, this is now ready for review! 🫡

File: CHANGELOG.md
Comment (ErichDonGubler) [inline CHANGELOG.md:1]: todo: @Dinnerbone was kind enough to [call out in Matrix](https://matrix.to/#/!FZyQrssSlHEZqrYcOb:matrix.org/$4k--nd7w8imaf4wCbrRyz1Z0dMparGzfhl0m9SON2Eg?via=mozilla.org&via=matrix.org&via=envs.net) that the new optionality of `entry_point` fields in pipeline descriptors is uniquitously breaking, and should feature as a major item.

Comment (ErichDonGubler) [reply CHANGELOG.md:1]: Fixed with 963136d1.

File: CHANGELOG.md
Comment (tychedelia) [inline CHANGELOG.md:189]: Looks like #6107 is missing here. We ran into this in bevy (see https://github.com/bevyengine/bevy/pull/15988#issuecomment-2439755263). It would be nice to call out needing to add `QuartzCore.framework` to any xcode builds.

File: CHANGELOG.md
Comment (Wumpf) [inline CHANGELOG.md]: typo:
```suggestion
The answer is: yes, we _do_ have sad, sad humans that really want their WGPU code to work _everywhere_. As Firefox and others use WGPU to implement WebGPU, the above example of Firefox diverging from standard is, unfortunately, today's reality. It _mostly_ behaves the same as a standards-compliant WebGPU, but it still doesn't in many important ways. Of particular note is Naga, its implementation of the WebGPU Shader Language. Shaders are pretty much a black-and-white point of failure in GPU programming; if they don't compile, then you can't use the rest of the API! And yet, it's extremely easy to run into this:
```

nit: for longer blocks like this more newlines (which are ignored in rendered markdown) are nicer for reviewing :)

File: CHANGELOG.md
Comment (Wumpf) [reply CHANGELOG.md:189]: thanks for calling this out! definitely one that shouldn't be missing 👍 

Btw. note that generally contributors are responsible for adding changelog notes (it's part of the tick boxes in the pr template) and reviewers should point that if missing (which in this case was... me, oops)

**Review by Wumpf — APPROVED** [review]
once more an excellent introduction, thank you so much!

File: CHANGELOG.md
Comment (ErichDonGubler) [reply CHANGELOG.md:189]: I'm not sure what content to write for this. @madsmtm and/or @Wumpf, can you please do that and file a PR against this PR's branch? I would also consider fixing this as follow-up, if it comes to that, since this is the only thing that might block merging. Tagging @gfx-rs/wgpu for opinions.

File: CHANGELOG.md
Comment (ErichDonGubler) [reply CHANGELOG.md]: RE: typo: fixed with latest force-push to [`78c1c8e5`](https://github.com/gfx-rs/wgpu/compare/06b37fa98f3c8e0b2a38ec6a1f95ad91d809f2d7..78c1c8e5d72d2bff3510e4db56010feaa9b787e6).

Comment (ErichDonGubler) [reply CHANGELOG.md]: RE: nit: Counter-nit (🙃): I deliberately coalesced this and existing content to be in single lines to match the style of the rest of the document. I would _really_ like to have soft breaks, because I think it'd make this document easier to deal with WRT diffs and editing. However, this is a big enough style thing that I think we should invoke an opinion from @gfx-rs/wgpu in a non-blocking fashion.

We can always change this formatting later automatically via a tool like `prettier`, so I'm not concerned about this blocking this PR.

File: CHANGELOG.md
Comment (Wumpf) [reply CHANGELOG.md:189]: I'll just push a commit into this branch if you don't mind

Comment (Wumpf) [reply CHANGELOG.md:189]: Hm doesn't allow to push me. I'd have written something like:

```
- Unless an explicit `CAMetalLayer` is provided, surfaces now render to a sublayer. This improves resizing behavior, fixing glitches during on window resize. By @madsmtm in [#6107](https://github.com/gfx-rs/wgpu/pull/6107).
```

Comment (madsmtm) [reply CHANGELOG.md:189]: Whoopsie, my bad. What @Wumpf has written sounds fine.

Comment (ErichDonGubler) [reply CHANGELOG.md:189]: @Wumpf: That's odd! You should be able to with something like `git push --force git@github.com:erichdongubler-mozilla/wgpu.git HEAD:v23` (N.B., not `--force-with-lease`). 🤔

Pushed myself with a6112742.

**Review by mnmaita — APPROVED** [review]

**Comment by ErichDonGubler** [comment]
@gfx-rs/wgpu, @gfx-rs/naga: I am now starting the final steps of the release, with the intent to publish to Crates.io immediately after automatic rebase-and-merge.

**Comment by ErichDonGubler** [comment]
D'oh, just need a review from @gfx-rs/naga.

**Review by nical — APPROVED** [review]
:rocket:

**Pull request merged**
